tools/Bridge/Bridge/JsonStringMediaTypeFormatter.cs

```diff
@@ -77,14 +75,48 @@ public override object ReadFromStream(Type type, Stream readStream, HttpContent
                 if (match.Success && match.Groups.Count == 2)
                     return new resource() { name = match.Groups[1].Value };
             }
-            else if (type == typeof(config))
+            else if (type == typeof(IDictionary<string,string>) || type.IsAssignableFrom(typeof(IDictionary)))
             {
-                var match = regexConfig.Match(data);
-                if (match.Success && match.Groups.Count == 2)
-                    return new config() { resourcesDirectory = match.Groups[1].Value };
+                return DeserializeDictionary(data);
             }
 
             return data;
         }
+
+        private static string SerializeDictionary(IDictionary dictionary)
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.AppendLine("{");
+            foreach (var key in dictionary.Keys)
+            {
+                sb.AppendFormat("   {0} : \"{1}\",\n", key, dictionary[key] == null ? String.Empty : dictionary[key].ToString());
+            }
+
+            sb.Remove(sb.Length - 2, 2);
+            sb.AppendLine("\n}");
+            return sb.ToString();
+        }
+
+        private static Dictionary<string, string> DeserializeDictionary(string data)
+        {
```

Comment:
Reviewers: I'm not happy parsing json here but had issues trying to get the built-in WebAPI json media type formatter to work properly.  To make progress, I would like to check-in this manual parsing and later convert to normal WebAPI json deserialization (issue #214).